cocotb/share/makefiles/Makefile.sim

```diff
@@ -79,7 +79,19 @@ ifeq ($(MAKECMDGOALS),help)
     help_vars := $(subst %,${newline},$(shell cocotb-config --help-vars | tr \\n %))
     $(info ${help_vars})
     # is there a cleaner way to exit here?
-    $(error "Stopping after printing help")
+    $(error Stopping after printing help)
+endif
+
+# Safety net to bail out early if users go into this Makefile without making
+# cocotb-config available in their PATH.
+# Use override to prevent users from overriding COCOTB_CONFIG_CMD, which is not
+# supported. (Modify PATH instead.)
+override COCOTB_CONFIG_CMD := $(shell :; command -v cocotb-config)
+ifeq ($(COCOTB_CONFIG_CMD),)
+    $(error cocotb-config not found! Cocotb is not installed (or is installed \
+            but not on the PATH). \
+            Please refer to https://docs.cocotb.org/en/latest/install.html \
```

Comment:
This always points to "latest" - add the check for "dev" in the version string that I had in an older Makefile.sim?

Reply:
Not convinced it's worth for a message which is very rarely shown, but why not. Do you have a pointer to the code you're referring to?

Reply:
https://github.com/cocotb/cocotb/blame/a253ee28df6f003421eedda71133b30de5c49033/cocotb/share/makefiles/Makefile.sim#L99 ff

Reply:
That DOCLINK code moved to cocotb-config now BTW - maybe (just maybe) we should even have a central place or function for it.

Reply:
That's unfortunately not going to work, since this piece of code is about finding `cocotb-config` in the first place. I'd just leave it as-is.

Reply:
If someone has old include structure (both first `Makefile.inc` and then `Makefile.sim`) would this not make issue since `COCOTB_CONFIG_CMD` is not defined for when the first include is `Makefile.inc`?